scripts/run-bundled-codegen.sh

```diff
@@ -99,8 +99,13 @@ download_apollo_cli_if_needed
 # Make sure we're using an up-to-date and valid version of the Apollo CLI
 validate_codegen_and_extract_if_needed
 
-# Add the binary directory to the beginning of PATH so included binary verson of node is used.
-PATH="${SCRIPT_DIR}/apollo/bin:${PATH}"
+# Give Apple Silicon users the option to install & use an ARM-based binary of node through Homebrew
```

Comment:
Would this be better as this?

```
 # Add the binary directory to the beginning of PATH so included binary verson of node is used.
 PATH="${SCRIPT_DIR}/apollo/bin:${PATH}"

# Give Apple Silicon users the option to install & use an ARM-based binary of node through Homebrew
if [[ `uname -m` == 'arm64' ]]; then
  PATH="/opt/homebrew/bin:${PATH}
fi
```
Seems cleaner and doesn't repeat parts of the path.

Reply:
@AnthonyMDev 

I think a better approach would be to extend the current bundling strategy. Apollo already bundles a binary for apollo-cli in it's Pod (`${SCRIPT_DIR}/apollo/bin`), so I think it makes sense to just bundle another binary (lets call it `${SCRIPT_DIR}/apollo/bin-arm64` for clarity's sake) and check architecture and define `PATH` based on your proposed solution.

This would keep in place the solution to avoid dependency hell with NPM and allow for Apple Silicon to work natively and out of the box, without having to figure out if users have homebrew, npm,  yarn, or whatever else they might (or might not) have.

```
PATH="${SCRIPT_DIR}/apollo/bin:${PATH}
if [[ `uname -m` == 'arm64' ]]; then
  PATH=${SCRIPT_DIR}/apollo/bin-arm64:${PATH}
fi
```

Reply:
@aravasio Apollo currently doesn't prepackage any binaries in the release, @designatednerd explained in #1611 they download the necessary source files to build the `node` (x86_64) binary for usage. 

So while I agree that from a pragmatic & long term POV building the ARM64 binary of the library as well would be better, given the fact that the upcoming `1.0` release is going to move away from `node` completely, the effort of incorporating & maintaining the ARM64 build of `node` could outweigh the benefits.

Taking into account that the largest developer userbase on macOS is likely already using & acquainted with Homebrew, the currently proposed solution is merely a minimally intrusive workaround until `1.0` is released. 

@AnthonyMDev I like your solution better indeed, I'll adapt the current proposal.